src/resolvers/authChecker.ts

```diff
@@ -0,0 +1,22 @@
+import { AuthChecker } from 'type-graphql';
+
+import { Context } from '@/resolvers/types';
+
+export const authChecker: AuthChecker<Context> = ({ context: { me } }, roles) => {
+  if (roles.length === 0) {
```

Comment:
Perhaps we can refer to the https://github.com/Canner/graphql-rbac for the role authentication and break this function into two part; 1. Checking if user login 2. Checking the Role and permission.